### PR TITLE
Converts and Updates Weather Subsystem

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over TG's weather system to a subsystem and updates it their latest standard.

Also converts over the radiation storm event to be a weather type instead of a snowflakey event.

The amount of work involved in getting this working was downright stupid.

Changes
- Removed advanced darkness (this has always been horrifically broken and requires lots of fiddling to get it to accomplish what you want it to, and it breaks area updates forever)
- Ash storm has better sounds now, with special sounds that play indoors and outdoors

:cl: Fox McCloud
tweak: Radiation event reworked; graphics updated--effects may be a bit more deadly
/:cl:

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -197,13 +201,18 @@
...
+			if(W.stage != END_STAGE && (src in W.impacted_areas))
+				W.update_areas()
+				weather_icon = TRUE
+		if(!weather_icon)
+			icon_state = null
+			invisibility = INVISIBILITY_MAXIMUM
 
 /area/space/updateicon()
 	icon_state = null
-	invisibility = INVISIBILITY_MAXIMUM
```
Comment (Fox-McCloud) [inline code/game/area/areas.dm]: re-add

File: code/modules/space_management/level_traits.dm [hunk 2]
```diff
@@ -1,62 +1,63 @@
...
+	var/list/trait_list
+	if(space_manager.initialized)
+		var/datum/space_level/S = space_manager.get_zlev(z)
+		trait_list = S.flags
+	else
+		trait_list = default_map_traits[z]
+		trait_list = trait_list["attributes"]
+	return (trait in trait_list)
 
 /proc/levels_by_trait(trait)
```
Comment (Fox-McCloud) [inline code/modules/space_management/level_traits.dm]: actual changes here; it just returns a list of the numerical z-levels instead of the space levels themselves; this was always used for the numerical z values, and never to return the space level datums.

File: code/game/machinery/computer/pod.dm [hunk 3]
```diff
@@ -205,8 +205,8 @@
 		if(href_list["dstele"])
 			var/choices = list(0)
 			var/list/reachable_levels = levels_by_trait(REACHABLE)
-			for(var/datum/space_level/S in reachable_levels)
```
Comment (tigercat2000) [inline code/game/machinery/computer/pod.dm]: Why did you change this? levels_by_trait returns a list of space_level datums.

Comment (Fox-McCloud) [reply code/game/machinery/computer/pod.dm]: See my above comment, and note what the code actually does.

It returns a list of `/datum/space_level`, adds their `zpos` (numerical z value) to a list, then promptly says "who cares" and discards the list. 

Anywhere and everywhere `levels_by_trait` is utilized, it always returns a list of space levels, then ends up being used for their z-value anyway.

No point in having extra code outside of a proc if this is what it's actually being used for literally 100% of the time

Comment (tigercat2000) [reply code/game/machinery/computer/pod.dm]: o i c

**Comment by Fox-McCloud** [comment]
Reverted the plane changes, except one.

`HUD_PLANE` has been lowered from 90 to 19. When this plane was added, it was before most of the other defined were added; as far as I can tell, the number is arbitrary.

In any event, it's still the highest plane number ***actively in use*** so I think it should be fine.

File: code/modules/events/radiation_storm.dm [hunk 4]
```diff
@@ -1,70 +1,11 @@
...
 /datum/event/radiation_storm/start()
-	spawn()
-		event_announcement.Announce("High levels of radiation detected near the station. Please evacuate into one of the shielded maintenance tunnels.", "Anomaly Alert", new_sound = 'sound/AI/radiation.ogg')
-
-		for(var/area/A in world)
-			if(!is_station_level(A.z) || is_safe_zone(A))
-				continue
-			A.radiation_alert()
-
-		make_maint_all_access()
```
Comment (tigercat2000) [inline code/modules/events/radiation_storm.dm]: You should probably readd this.
